packages/typescript-estree/src/create-program/createWatchProgram.ts

```diff
@@ -309,25 +309,30 @@ function createWatchProgram(
       );
     oldOnDirectoryStructureHostCreate(host);
   };
-
-  /*
-   * The watch change callbacks TS provides us all have a 250ms delay before firing
-   * https://github.com/microsoft/TypeScript/blob/b845800bdfcc81c8c72e2ac6fdc2c1df0cdab6f9/src/compiler/watch.ts#L1013
-   *
-   * We live in a synchronous world, so we can't wait for that.
-   * This is a bit of a hack, but it lets us immediately force updates when we detect a tsconfig or directory change
-   */
-  const oldSetTimeout = watchCompilerHost.setTimeout;
-  watchCompilerHost.setTimeout = (cb, ms, ...args): unknown => {
-    if (ms === 250) {
-      cb();
-      return null;
+  watchCompilerHost.trace = log;
+
+  // Since we don't want to asynchronously update program we want to disable timeout methods
+  // So any changes in the program will be delayed and updated when getProgram is called on watch
+  // But because of https://github.com/microsoft/TypeScript/pull/37308 we cannot just set it to undefined
+  // instead save it and call before getProgram is called
+  let callback: (() => void) | undefined;
```

Comment:
So this is introduced for improving pre-TS 3.9 scenarios - does it still make sense when running with https://github.com/microsoft/TypeScript/pull/37308?

Reply:
Till we have the release build with that fix, this is the only solution. After that we add condition to just unset these timeout methods for that release onwards .. But both those are essentially almost same// that is do not immediately run those callbacks but wait till someone asks for updated program...